<img src="https://i.ibb.co/93sNcdhp/Logo-pmat-color.png" align="right" width="150">

<br>

# **Laboratorio 0 - Estadística Descriptiva y bases inferenciales**

## **Objetivos de aprendizaje**

Al finalizar este laboratorio, el estudiante pondrá en práctica sus habilidades para:
* Manipular bases de datos reales utilizando librerías de Python.
* Caracterizar una población a través de análisis descriptivo.
* Comparar la distribución observada de los datos con el modelo teórico Normal.
* Comprender la variabilidad muestral mediante simulación.
* Reconocer cómo estos conceptos se conectan con los métodos de inferencia estadística.

## **Recordatorio clave**

Antes de iniciar los contenidos de inferencia es fundamental recordar:
* **Media y Mediana:** describen el comportamiento central de los datos.
* **Desviación estándar:** base del error estándar y variabilidad.
* **Proporciones:** fundamentales para inferencia categórica.
* **Normalidad:** fundamento de las probabilidades, valores críticos y muchos métodos inferenciales.
* **Variabilidad muestral:** las muestras cambian → por eso existe la inferencia.
* **Parámetro vs Estadístico:**

  *Parámetro:* valor desconocido de la población ($\mu$, $\sigma$, $p$).

  *Estadístico:* valor calculado en una muestra ($ \bar{x}$, $s$, $\hat{p}$).

  *Inferencia* = usar un estadístico para estimar un parámetro.
* **Python:** herramienta para cálculo, visualización y simulación.

# **Contexto General**

La Dirección de una Institución de educación superior está desarrollando un análisis descriptivo preliminar para comprender los factores que podrían estar relacionados con el rendimiento académico en asignaturas críticas. Para ello se construyó un dataset de 300 estudiantes de distintas carreras, que contiene las siguientes variables: carrera, nota_certamen, asistencia, satisfaccion, usa_python, horas_estudio, ingreso_anual, sector_laboral, nivel_estudios, promedio_final.

Antes de aplicar métodos inferenciales, es necesario revisar los conocimientos previos fundamentales para interpretar los resultados que se obtendrán a lo largo del semestre.

**Instrucciones**
* Lee atentamente cada contexto y responde las preguntas planteadas.
* Utiliza Python cuando sea necesario para realizar cálculos.
* Algunas preguntas requieren solo análisis conceptual y otras requieren cálculo.
* Descarga el **[dataset](https://docs.google.com/spreadsheets/d/1D76FMmq2SEPDpBkOL4BNsHTQ3LJeMEZJ/export?format=xlsx)** para que puedas cargarlo en tu entorno de trabajo.

In [24]:
import pandas as pd
import glob
from scipy.stats import skew, norm, kurtosis

In [2]:
glob.glob('*')

['G0_A2_estudiantes.xlsx', 'Laboratorio_0_MAT4152.ipynb']

In [3]:
df = pd.read_excel('G0_A2_estudiantes.xlsx')

In [6]:
df.describe()

,nota_certamen,asistencia,satisfaccion,usa_python,horas_estudio,ingreso_anual_US,promedio_final
count,300.000000,300.000000,300.000000,300.000000,300.00000,300.000000,300.000000
mean,5.035000,80.244967,3.136667,0.680000,11.88300,64345.143333,4.962333
std,0.808613,10.671084,1.167275,0.467256,3.64367,26417.589255,1.010024
min,2.600000,54.720000,1.000000,0.000000,5.00000,13120.000000,1.800000
25%,4.500000,73.110000,2.000000,0.000000,9.27500,45256.000000,4.300000
50%,5.100000,80.470000,3.000000,1.000000,11.85000,60180.500000,5.000000
75%,5.600000,87.850000,4.000000,1.000000,14.50000,78691.250000,5.700000
max,7.000000,100.000000,5.000000,1.000000,24.90000,183236.000000,7.000000


In [ ]:
var = 'nota_certamen'
diccionario = {
    'Cantidad': df[var].count(),
    'Media': df[var].mean(),
    'Mediana': df[var].median(),
    'DesvStand': df[var].std(),
    'Curtosis': kurtosis(df[var]),
    'Asimetria': skew(df[var]),
    'Moda': df[var].mode(),
    'Min': df[var].min(),
    'P10': df[var].quantile(0.10),
    'P25': df[var].quantile(0.25),
    'P50': df[var].quantile(0.50),
    'P75': df[var].quantile(0.75),
    'P90': df[var].quantile(0.90),
    'Max:': df[var].max(),
    'Aprobados': (df[var] >= 4).mean()*100,
    'Reprobados': (df[var] <= 4).mean()*100
}

pd.DataFrame(diccionario, index=[0])

,Cantidad,Media,Mediana,DesvStand,Curtosis,Asimetria,Moda,Min,P10,P25,P50,P75,P90,Max:,Aprobados,Reprobados
0,300,5.035,5.1,0.808613,-0.193682,-0.217382,5.2,2.6,4.0,4.5,5.1,5.6,6.01,7.0,90.666667,12.666667


### **Mini-activación diagnóstica**

Antes de avanzar, responde brevemente:

* Clasifica las variables: carrera, ingreso_anual, asistencia.

In [26]:
df.sample(3)

,carrera,nota_certamen,asistencia,satisfaccion,usa_python,horas_estudio,ingreso_anual_US,sector_laboral,nivel_estudios,promedio_final
240,Arquitectura,5.6,66.69,4,0,11.0,66873,Comercio,Medio,5.3
141,Psicología,5.1,78.93,3,1,16.6,87956,Comercio,Medio,5.1
48,Enfermería,5.1,87.03,3,1,9.7,84943,Educación,Medio,6.0


In [ ]:
cols = ['carrera', 'sector_laboral']
df[cols].drop_duplicates(keep='first')

,carrera,sector_laboral
0,Derecho,Salud
1,Arquitectura,Servicios
3,Psicología,Educación
4,Enfermería,Servicios
5,Derecho,Tecnología
6,Enfermería,Comercio
7,Psicología,Salud
8,Psicología,Servicios
9,Ingeniería,Salud
12,Ingeniería,Educación


In [33]:
for col in cols:
    print(df[col].unique())

['Derecho' 'Arquitectura' 'Psicología' 'Enfermería' 'Ingeniería']
['Salud' 'Servicios' 'Educación' 'Tecnología' 'Comercio']


Carrera: Variable Cualitativa Nominal con valores unicos ['Derecho' 'Arquitectura' 'Psicología' 'Enfermería' 'Ingeniería']
Ingreso Anual: Variable Cuantitativa Continua
Asistencia: Variable Cuantitativa Continua

* ¿Cuál es la diferencia entre media y mediana?

Ambas con medidas de tendencia central, pero la mediana separa el 50% de los datos (ordenados de menor a mayor) y la media es el promedio de los valores de la variable.|

* ¿Qué significa que la desviación estándar sea alta?

Indica una mayor dispersión de los datos con respecto a la media. Mayor variabilidad y menor homogeneidad.

# **1. Clasificación de variables**

A partir del dataset, clasifica las siguientes variables:

a) Nivel de estudios

b) Ingreso Anual

c) Carrera

d) Promedio Final

# **2. Medidas descriptivas básicas**

2.1. Considera la variable **nota_certamen** del dataset.

a) Calcula la media.


In [37]:
var = 'nota_certamen'
print(f"La media de {var} es {df[var].mean():.2f}")

La media de nota_certamen es 5.04


b) Calcula la mediana.

In [38]:
print(f"La mediana de {var} es {df[var].median():.2f}")

La mediana de nota_certamen es 5.10


c) ¿Estas dos medidas son similares o distintas? ¿Qué sugiere esto del comportamiento de la variable?

**Los valores son muy cercanos, pero la mediana  la derecha de la media indica presencia de valores atípicos en el lado derecho de la variable**

d) Explica qué representaría un valor muy bajo de mediana en este contexto.

**Representaría que más del 50% de los alumnos obtuvo una nota muy baja menor al promedio del curso**

2.2. Usa la variable **asistencia**.

a) Calcula la media y la mediana

In [40]:
var = 'asistencia'
print(f"La media de {var} es {df[var].mean():.2f}")
print(f"La mediana de {var} es {df[var].median():.2f}")

La media de asistencia es 80.24
La mediana de asistencia es 80.47


b) Compara ambos valores y explica qué implicaría una mediana muy alta en este contexto.

**Los valores son muy cercanos, pero si la mediana fuese muy alta, indicaría presencia de valores atípicos, señalando que más del 50% de los alumnos tuvo muy buena asistencia**

# **3. Dispersión del rendimiento**

3.1. Considera nuevamente la variable **nota_certamen**.

a) Calcula la desviación estándar muestral.

In [42]:
var = 'nota_certamen'
print(f"La desviación estándar muestral de {var} es {df[var].std():.2f}")

La desviación estándar muestral de nota_certamen es 0.81


In [44]:
df[var].std() / df[var].mean()

np.float64(0.16059835355929453)

b) Interpreta el resultado en el contexto del rendimiento académico.

**La desviación estandar es baja respecto a la media lo que nos indica que los datos tienen baja variabilidad y por ende una alta homogeneidad**

3.2. Usa la variable **horas_estudio**.

a) Calcula la desviación estándar.

In [45]:
var = 'horas_estudio'
print(f"La desviación estándar de {var} es {df[var].std():.2f}")

La desviación estándar de horas_estudio es 3.64


In [47]:
rango = df[var].max() - df[var].min()
print(f"El rango de {var} es {rango}")

El rango de horas_estudio es 19.9


b) Interpreta la desviación estándar considerando el rango de los datos e indica si los estudiantes estudian de manera homogénea o muy diferente entre sí.

In [48]:
df[var].std() / rango

np.float64(0.18309900213345415)

**Existe una variabilidad que se comporta de forma homogenea respecto al rango de horas de estudio**

# **4. Proporciones y probabilidad simple**

4.1 Usando la variable `usa_python`:

En la base de datos, la variable `usa_python` está codificada como: 1 = usa Python y 0 = no usa Python.

a) Estima la proporción de estudiantes que utilizan Python.

In [49]:
var = 'usa_python'
df[var].sum() / df[var].count()

np.float64(0.68)

In [57]:
f1 = df[var] == 1
prop = f1.mean()
print(f"La proporción de estudiantes que usa Python es {prop*100:.0f}%")

La proporción de estudiantes que usa Python es 68%


b) Interpreta qué significa esa proporción para la asignatura.

**El 68% de los estudiantes utiliza Python**

c) ¿Cuál es la probabilidad de seleccionar un estudiante que no usa Python?

In [61]:
f1 = df[var] == 1
prop = (~f1).mean()
print(f"La proporción de seleccionar un estudiante que no usa Python es {prop*100:.0f}%")

La proporción de seleccionar un estudiante que no usa Python es 32%


4.2. Usa la variable `satisfaccion`.

a) Calcula la proporción de estudiantes que están en niveles de satisfacción 4 o 5. Considera que el nivel de satisfacción se mide desde 1 a 5.

In [68]:
var = 'satisfaccion'
f1 = df[var] >= 4
prop = f1.mean()
print(f"La proporción de estudiantes con satisfacción entre 4 y 5 es {prop*100:.1f}%")

La proporción de estudiantes con satisfacción entre 4 y 5 es 38.3%


b) Interprétalo para la institución.

**El 38.33% delos estudiantes está satisfecho o muy satisfecho con la institución**

c) Calcula la probabilidad de seleccionar un estudiante con satisfacción menor o igual a 2.

# **5. Evaluación descriptiva del desempeño final**

5.1 Usa la variable `horas_estudio`:

a) Calcula asimetría y curtosis.

b) Calcula las medidas de tendencia central.

c) Construye el histograma para la variable.

c) ¿La distribución de horas de estudio es aproximadamente simétrica?

5.2. Usa la variable `promedio_final`.

a) La variable que se describe, ¿se distribuye normalmente? ¿Por qué? Justifique analizando las medidas de distribución, el gráfico correspondiente y las medidas de tendencia central.

# **6. Distrubución normal**


6.1. Asumiendo que la variable `promedio_final` distribuye aproximadamente normal, determine:

a) ¿Cuál es la probabilidad de que un estudiante obtenga un promedio final mayor a 5,5?

b) ¿Cuál es la probabilidad de que un estudiante obtenga un promedio final menos a 4,0?

c) ¿Cuál es la probabilidad de que un estudiante obtenga un promedio final entre a 4,5 y 6,5?

6.2. Asumiendo que la variable `horas_estudio` distribuye aproximadamente normal, determine su media y desviación estándar. Luego, responda las preguntas indicadas.

a) ¿Cuál es la probabilidad de que un estudiante dedique menos de 10 horas a estudiar?

b) ¿Cuál es la probabildiad de que un estudiante dedique entre 15 y 20 horas a estudiar?

c) ¿Cuál es la probabilidad de que un estudiante dedique más de 11 horas a estudiar?

# **7. Comparación de grupos**

7.1 Divide los estudiantes según la variable `usa_python`:

a) Calcula la desviación estándar, la media y el coeficiente de variación de `horas_estudio` en ambos grupos.



b) Describe e interpreta las diferencias observadas entre ambos grupos, considerando la media y el coeficiente de variación de las horas de estudio. ¿Qué sugieren estos resultados sobre los hábitos de estudio de los estudiantes que usan Python y los que no?

7.2. Divide a los estudiantes según `nivel_estudios` (Básico / Medio / Superior)

a) Calcula la media, la desviación estándar y el coeficiente de variación de la asistencia para cada nivel de estudios.

b) Describe e interpreta las diferencias observadas entre los niveles de estudios

# **8. Variabilidad muestral**

En estadística inferencial, la variabilidad muestral explica por qué las estimaciones pueden cambiar cuando se trabaja con distintas muestras extraídas de una misma población.

Para explorar este concepto, trabajaremos con la variable `promedio_final`.

Descarga **[aquí](https://docs.google.com/spreadsheets/d/1axWLVym_wXsVSjKl0saE7-T3hGycdJ8Y/export?format=xlsx)** el dataset que contiene 100 muestras, cada una de tamaño $n=20$, obtenidas a partir del dataset original.

a) Carga el archivo con las 100 muestras.

b) Calcula la media de cada muestra, obteniendo así 100 medias muestrales.

c) Grafica un histograma de las 100 medias muestrales e interpreta su forma.

d) ¿Las medias muestrales son iguales entre sí?

e) Compara la media del curso con el promedio de las 100 medias muestrales.

f) ¿A qué atribuyes que estos valores son similares pero no exactamente iguales?

# **Con este laboratorio has reforzado los siguientes conceptos:**

* Clasificar variables en un contexto aplicado, identificando correctamente su tipo (cualitativas y cuantitativas) y su rol en el análisis estadístico.
* Calcular e interpretar medidas descriptivas, como la media, la desviación estándar y el coeficiente de variación, para resumir y comprender conjuntos de datos reales.
* Estimar proporciones y probabilidades simples, vinculando los resultados con situaciones concretas y su interpretación en contexto.
* Analizar el comportamiento de una variable con distribución normal, reconociendo sus principales características y su utilidad en la modelación de fenómenos reales.
* Calcular probabilidades, comprendiendo qué tan frecuente o probable es un evento aleatorio.
* Explorar relaciones entre variables, utilizando herramientas descriptivas y gráficas para identificar patrones o asociaciones.
* Introducirte al concepto de variabilidad muestral y comprender su relevancia, entendiendo por qué las estimaciones cambian entre muestras y cómo se relacionan con el valor poblacional.
* Utilizar Python como herramienta de análisis estadístico, aplicándolo para calcular, visualizar e interpretar información de manera eficiente y reproducible.
